features/filesystem/fat/FATFileSystem.cpp

```diff
@@ -333,24 +333,69 @@ int FATFileSystem::unmount()
 int FATFileSystem::format(BlockDevice *bd, bd_size_t cluster_size)
 {
     FATFileSystem fs;
-    int err = fs.mount(bd, false);
+    fs.lock();
+
+    int err = bd->init();
     if (err) {
+        fs.unlock();
+        return err;
+    }
+
+    // erase disk
+    err = bd->erase(0, bd->size());
```

Comment:
Will this solution work for all block devices? In case of SD we do not implement erase, erase is done as part of trim.